tasks/test_results_processor.py

```diff
@@ -433,6 +471,15 @@ def process_individual_upload(
             extra=dict(upload_id=upload_id),
         )
 
+        upload_obj.state = "processed"
+        db_session.commit()
+
+        readable_report = self.rewrite_readable(
+            arg_processing_result.network_files,
+            arg_processing_result.report_contents,
+        )
+        archive_service.write_file(upload_obj.storage_path, readable_report.getvalue())
```

Comment:
Didn't realize you couldn't change states before committing 😮.

Since I've been reading the coverage processor code lately, what are your thoughts on saving intermediate states of these in redis and having the finisher save them in bulk to the DB? Chatting w/ Arpad, it seemed to be to help hammering the DB as frequently, idk if this could benefit from that? Food for thought

Reply:
> saving intermediate states of these in redis and having the finisher save them in bulk to the DB
probably a good idea honestly, writes to redis is probably cheaper than a bunch of transactions with updates 
writing to the same keys